src/main/java/products/Stock.java

```diff
@@ -0,0 +1,44 @@
+package products;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Stock {
+
+    private Map<Categories, Category> categories = new HashMap<>();
+    private List<Product> products = new ArrayList<>();
+
+    private List<Category> makeCategoryList(Categories[] category) {
+        List<Category> categoriesOfProduct = new ArrayList<>();
+        for(Categories current: category) {
+            categoriesOfProduct.add(categories.get(current));
```

Comment:
what if we join it with 
```
for(Categories current: category) {
    if(!categories.containsKey(current)) {
        categories.put(current, new Category(current));
    }
}
```
So it could be something like:

```
for(Categories current: category) {
	Category category = categories.get(current);
    if(category == null) {
        category = new Category(current);
        categories.put(current, category);
    }
    categoriesOfProduct.add(category);
}
return categoriesOfProduct;
```